public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation
+} from '@angular/core';
+import { BaseReactWrapper } from 'app/data-browser/base-react/base-react.wrapper';
+import { baseOptions } from 'app/data-browser/charts/react-base-chart/base-chart.service';
+import * as highCharts from 'highcharts';
+import HighchartsReact from 'highcharts-react-official';
+import * as React from 'react';
+
+interface State {
+    options: any;
+    ageAnalysis: any;
+    ageCountAnalysis: any;
+}
+
+interface Props {
+    ageAnalysis: any;
+    ageCountAnalysis: any;
+    selectedResult: any;
+    domain: string;
+}
+
+export class AgeChartReactComponent extends React.Component<Props, State> {
+
+  constructor(props) {
+    super(props);
+    this.state = {options: null, ageAnalysis: props.ageAnalysis,
+    ageCountAnalysis: props.ageCountAnalysis};
+  }
+
+  componentDidMount() {
+    this.getChartOptions();
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>) {
+    const newAnalysis = this.props.ageAnalysis;
+    if (newAnalysis !== this.state.ageAnalysis) {
+        this.setState({ ageAnalysis: newAnalysis }, this.getChartOptions);
+    }
```

Comment:
We can compare current props with `prevProps` arg to check for changes:
```suggestion
    if (prevProps.ageAnalysis !== this.props.ageAnalysis) {
        this.getChartOptions();
    }
```